pkg/tools/etcd_tools.go

```diff
@@ -359,10 +409,11 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 			return nil
 		}
 
-		_, err = h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
+		response, err := h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
```

Comment:
Will allow create on update, but not overwrite on update (but that can be done from the client).  There's no way to do a blind put today, so it resolves one of those use cases but not the other.

----- Original Message -----
> > @@ -359,10 +409,11 @@ func (h *EtcdHelper) AtomicUpdate(key string,
> > ptrToType runtime.Object, ignoreNo
> >  			return nil
> >  		}
> >  
> > -		_, err = h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
> > +		response, err := h.Client.CompareAndSwap(key, string(data), 0, origBody,
> > index)
> 
> Is #656 still unresolved?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4419/files#r24681444